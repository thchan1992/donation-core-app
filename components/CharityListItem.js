import { TouchableOpacity, View, Text, StyleSheet, Image } from "react-native";

const CharityListItem = ({ name, imageUrl, onPress }) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Image
        style={{
          height: 150,
          width: 150,
          alignSelf: "center",
          justifyContent: "center",
          borderRadius: 8,
          borderWidth: 1.5,
          borderColor: "#FF304F",
        }}
        source={imageUrl}
      />
      <View>
        <Text
          style={[
            styles.text,
            name.length > 10 ? styles.smallerText : styles.text,
          ]}
        >
          {name}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default CharityListItem;

const styles = StyleSheet.create({
  button: {
    width: 150,
    height: 180,
    margin: 10,
    borderRadius: 8,
    borderWidth: 1.5,
    backgroundColor: "#28C7FA",
    borderColor: "#FF304F",
  },
  text: {
    alignSelf: "center",
    padding: 3,
    fontWeight: "bold",
    textTransform: "uppercase",
    fontSize: 20,
    color: "black",
  },

  smallerText: {
    alignSelf: "center",
    padding: 5,
    fontWeight: "bold",
    textTransform: "uppercase",
    fontSize: 12,
    color: "black",
  },
});
