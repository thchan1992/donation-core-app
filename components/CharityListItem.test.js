import renderer from "react-test-renderer";
import React from "react";
import CharityListItem from "./CharityListItem";
import { TouchableOpacity, Text } from "react-native";

describe("CharityListItem", () => {
  let onPress;

  beforeEach(() => {
    onPress = jest.fn();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <CharityListItem
          name="Test Charity"
          imageUrl={require("../assets/adaptive-icon.png")}
          onPress={onPress}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const component = renderer.create(
      <CharityListItem
        name="Test Charity"
        imageUrl={require("../assets/adaptive-icon.png")}
        onPress={onPress}
      />
    );
    const instance = component.root;
    const button = instance.findByType(TouchableOpacity);
    button.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
