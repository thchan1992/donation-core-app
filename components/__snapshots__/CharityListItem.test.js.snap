// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityListItem renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#28C7FA",
      "borderColor": "#FF304F",
      "borderRadius": 8,
      "borderWidth": 1.5,
      "height": 180,
      "margin": 10,
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "#FF304F",
        "borderRadius": 8,
        "borderWidth": 1.5,
        "height": 150,
        "justifyContent": "center",
        "width": 150,
      }
    }
  />
  <View>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
            "padding": 3,
            "textTransform": "uppercase",
          },
          Object {
            "alignSelf": "center",
            "color": "black",
            "fontSize": 12,
            "fontWeight": "bold",
            "padding": 1,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Test Charity
    </Text>
  </View>
</View>
`;
